plonk/src/circuit/customized/ecc/mod.rs

```diff
@@ -263,12 +263,18 @@ where
         &mut self,
         point0: &PointVariable,
         point1: &PointVariable,
-    ) -> Result<Variable, PlonkError> {
+    ) -> Result<BoolVar, PlonkError> {
         self.check_point_var_bound(point0)?;
         self.check_point_var_bound(point1)?;
         let x_eq = self.check_equal(point0.0, point1.0)?;
         let y_eq = self.check_equal(point0.1, point1.1)?;
-        self.mul(x_eq, y_eq)
+
+        let res = self.create_bool_variable_unchecked(
+            self.witness(x_eq.into())? * self.witness(y_eq.into())?,
+        )?;
+        self.mul_gate(x_eq.into(), y_eq.into(), res.into())?;
```

Comment:
Can we replace `self.mul(x_eq, y_eq)` (in the olde commit) with `self.logic_and(x_eq: BoolVar, y_eq : BoolVar) -> BoolVar`? We can simplify the code here by doing that.

Reply:
oh yes, good suggestion

Reply:
done in [5f0cd66](https://github.com/EspressoSystems/jellyfish/pull/94/commits/5f0cd663417b46145cedbd3f620aa8963808ba00)